File: lib/utils/animations/entrance/flip3d_y.dart
import '../../base/animated_text_state.dart';

AnimatedTextState evaluateFlip3DY(double t) {
  final rotationY = -90.0 * (1.0 - t);
  final scale = 0.6 + 0.4 * t;
  return AnimatedTextState(
    opacity: t.clamp(0.0, 1.0),
    rotationY: rotationY,
    scale: scale,
  );
}

File: android/app/src/main/kotlin/com/typography/animations/entrance/Flip3D_Y.kt
package com.typography.animations.entrance

import com.typography.AnimatedTextState

fun evaluateFlip3DY(t: Float): AnimatedTextState {
    val rotationY = -90.0f * (1.0f - t)
    val scale = 0.6f + 0.4f * t
    return AnimatedTextState(
        opacity = t.coerceIn(0f, 1f),
        rotationY = rotationY,
        scale = scale
    )
}
